### DISCUSSION TITLE
Round GeoDataFrame Coordinates

### DISCUSSION THREAD

**Discussion by bsaid97** [geopandas/geopandas#2871] [Q&A]
I'm reading a geojson file using `from_file` and then dissolving. All this works well, however I'm trying to round (using the `round` method) the bring down the coordinates to 5 decimal places. 

```
import geopandas as gpd
shape = gpd.read_file('depotArea/depot.json')
dissolved_shape = shape.dissolve()
dissolved_rounded = dissolved_shape.round(5)
dissolved_rounded.to_file('depotArea/dissolved.geojson', driver='GeoJSON')
```

I can't seem to get it rounded to any value I try. Am I doing something wrong?

**Comment by theroggy** [comment]
Rounding the coördinates of geometries is (as far as I know) not possible using the standard pandas round() function.

Because simply rounding all coordinates can lead to invalid geometries (self intersections,...), I would use a dedicated function that is meant to reduce the precision correctly for this rather than simply rounding them.

These functions aren't exposed in geopandas (yet), but they are available in shapely.

Something like this should do the trick (edited based on comments below from @martinfleis):
```
import shapely

dissolved_rounded = dissolved_shape.copy()
dissolved_rounded.geometry = shapely.set_precision(dissolved_shape.geometry.array, grid_size=0.00001)
```

**Reply by theroggy** [reply]
Remark: because st_precision is a shapely2 feature, possibly you need to add the following lines to your script BEFORE `import geopandas as gpd` to make sure that geometries are saved internally in shapely2 format by geopandas:
```
import os
os.environ['USE_PYGEOS'] = '0'
```

**Reply by martinfleis** [reply]
Or use 

```
dissolved_rounded.geometry.array
```

Instead of 

```
dissolved_rounded.geometry.array.data
```

That's compatible with both backends.

**Reply by theroggy** [reply]
@martinfleis 
```
dissolved_rounded.geometry.array = shapely.set_precision(dissolved_shape.geometry.array, grid_size=0.00001)
```
gives the following error: `AttributeError: can't set attribute 'array'`

Should it be like this? Look a bit weird?
```
dissolved_rounded.geometry.array.data = shapely.set_precision(dissolved_shape.geometry.array, grid_size=0.00001)
```

**Reply by martinfleis** [reply]
```
dissolved_rounded["geometry"]= shapely.set_precision(dissolved_shape.geometry.array, grid_size=0.00001)

```

This should work. Better not to assign to underlying arrays.

**Comment by i1yshk00** [comment]
try to change schema dtype 
docs: https://fiona.readthedocs.io/en/latest/manual.html#keeping-schemas-simple
